server.go

```diff
@@ -77,6 +77,20 @@ func Listen(conn net.PacketConn, config *Config) (Listener, error) {
 	}, nil
 }
 
+func populateConfig(config *Config) *Config {
+	versions := config.Versions
+	if len(versions) == 0 {
+		versions = make([]protocol.VersionNumber, len(protocol.SupportedVersions))
+		copy(versions, protocol.SupportedVersions)
+	}
+
+	return &Config{
+		TLSConfig: config.TLSConfig,
```

Comment:
Well, first of all, you can't copy a `tls.Config`, because it contains a mutex.
I'd like to have a function that takes a `quic.Config` provided by the user and populates all unset fields with default values, so that later on in the code we don't have to worry about that any more.
What do you think is the nicer way to do it? Create a new struct, effectively treating the user-supplied `quic.Config` as a `const`, or doing the replacements in that struct?